torch/ao/quantization/dynamic_tracing/quantization_state.py

```diff
@@ -0,0 +1,797 @@
+
+        return op, tuple(new_args), kwargs
+
+    def op_convert_after_hook(
+        self,
+        op: Callable,
+        output,
+    ) -> Any:
+        """
+        This function is called aftern an op call in a converted model.
+
+        TODO: add dequant, if needed
+        """
+        return output
+
+    def get_op_convert_info(
+        self,
+        op: Callable,
+        unwrap_scale_zp: bool = False,
+    ) -> Tuple[
+        Callable,
+        List[Optional[Tuple[float, int]]],
+        List[bool],
+        Optional[str],
+        Dict[str, Any]
+    ]:
+        """
+        Returns the information needed for convert time modifications to `op`.
+        Has no side effects.  Return types:
+
+        `new_op`. Returns either the original callable unchanged,
+        or the corresponding quantized target. Note: always returns the original
+        callable for modules, because they are quantized with module swaps.
+
+        `arg_quant_infos`. Returns information needed to quantize each arg, if
+        applicable.
+
+        `additional_kwargs`. Returns additional kwargs for scale and zero_point, if
+        applicable.
+        """
+        assert self.cur_op_needs_hooks(op)
+        seen_op = self._get_cur_seen_op()
+
+        # calculate new op
+        new_op = get_quantized_op(op, seen_op)
+
+        # calculate quant infos
+        arg_quant_infos, arg_dequant_infos = \
+            self._get_input_args_quant_dequant_info(op)
+
+        # get packed param name, if applicable
+        packed_param_name = self._get_packed_param_name(op)
+
+        # calculate scale and zp for output
+        # TODO: instead of always doing this if there is an observer,
+        # calculate whether this is needed based on the op and dtypes
+        additional_kwargs = {}
+        needs_scale_zp = converted_func_needs_scale_zp(op, seen_op)
+        if needs_scale_zp:
+            seen_op = self._get_cur_seen_op()
+            output_tensor_infos = seen_op.output_tensor_infos
+            tensor_id = output_tensor_infos[0].id
+            observer = self.tensor_id_to_observer[str(tensor_id)]
+
+            scale, zp = observer.calculate_qparams()
+            # TODO(future PR): remove this boolean flag
+            if not unwrap_scale_zp:
+                additional_kwargs.update({'scale': scale, 'zero_point': zp})
+            else:
+                additional_kwargs.update(
+                    {'scale': scale.item(), 'zero_point': zp.item()})
+
+        return new_op, arg_quant_infos, arg_dequant_infos, packed_param_name, additional_kwargs
+
+    def _get_packed_param_name(self, cur_op: Callable) -> Optional[str]:
+        """
+        If the op has a quantized packed param, returns it. Otherwise, returns
+        None.
+        """
+        return self.idx_to_packed_weight_name.get(str(self.idx), None)
+
+    # TODO(next): make this also return information about dequants, not just quants
+    def _get_input_args_quant_dequant_info(
```

Comment:
sorry, what do you mean?  I took a look at the link, not sure yet how it could help.  Is this doing something like

```
qparam_dict = get_qparam_dict(obs)
scale, zp = qparam_dict['scale'], qparam_dict['zp']
```

or something else?